packages/search-in-workspace/src/common/search-in-workspace-interface.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2017 Erisson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { JsonRpcServer } from "@theia/core";
+
+export interface SearchInWorkspaceOptions {
+        /**
+         * Maximum number of results to return.  Defaults to unlimited.
+         */
+        maxResults?: number;
+}
+
+export interface SearchInWorkspaceResult {
+        /**
+         * The path to the file containing the result.
+         */
+        file: string;
+
+        /**
+         * The (1-based) line number of the result.
+         */
+        line: number;
+
+        /**
+         * The (1-based) character number in the result line.  For UTF-8 files,
+         * one multi-byte character counts as one character.
+         */
+        character: number;
+
+        /**
+         * The length of the match, in characters.  For UTF-8 files, one
+         * multi-byte character counts as one character.
+         */
+        length: number;
+
+        /**
+         * The text of the line containing the result.
+         */
+        lineText: string;
+}
+
+/**
+ * Sort search in workspace results according to file, line, character position
+ * and then length.
+ */
+
+export function searchInWorkspaceResultCompare(a: SearchInWorkspaceResult, b: SearchInWorkspaceResult) {
```

Comment:
would be nice to have a namespace for it matching to the interface name:

```ts
export namespace SearchInWorkspaceResult {
    export function compare(...) {...}
}

Reply:
Is it possible to have both an interface and namespace with the same name?  How do we import both of them?

Reply:
Nevermind, it just seems to do the right thing.